### ISSUE TITLE
[Good First Issue]: Align behavior of ONNX Frontend function ReduceLogSumExp-11, 13, 18 with original framework

### ISSUE THREAD

**Issue by gkrivor** [openvinotoolkit/openvino#20562]
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceLogSumExp for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSumExp-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSumExp-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSumExp-18)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

First of all, please, take a look on [ReduceMax](https://github.com/openvinotoolkit/openvino/pull/23475) PR for a reference.

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceLogSumExp)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSumExp-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSumExp-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceLogSumExp-18)
1. Function already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/wPuqAujS) - engage in discussions, ask questions and talk to OpenVINO developers
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

**Comment by kshitij01042002** [comment]
Hi @gkrivor I would like to take up this issue, if it's available! Thanks.

**Comment by p-wysocki** [comment]
Hello @kshitij01042002, are you still working on that issue?

**Comment by mwilczy** [comment]
Hi, Started working on this,

<!DOCTYPE html>

<html>
<head>
	
	

	
</head>

<body>

  | ReduceLogSumExp-11 | ReduceLogSumExp-13 | ReduceLogSumExp-18
-- | -- | -- | --
Decription | Computes the log sum exponent of the input tensor's element along the provided axes. The resulting tensor has the same rank as the input if keepdims equals 1. If keepdims equal 0, then the resulted tensor have the reduced dimension pruned. | +Input tensors of rank zero are valid.Reduction over an empty set of values yields minus infinity (if supported by the datatype) or undefined otherwise. | NO CHANGE
Attributes | axes : list of intsA list of integers, along which to reduce. The default is to reduce over all the dimensions of the input tensor. Accepted range is [-r, r-1] where r = rank(data).keepdims : int (default is 1)Keep the reduced dimension or not, default 1 means keep reduced dimension. | NO CHANGE | +noop_with_empty_axes : int (default is 0)Defines behavior if 'axes' is empty. Default behavior with 'false' is to reduce all axes. When axes is empty and this attribute is set to true, input tensor will not be reduced,and the output tensor would be equivalent to input tensor.-axes : list of intsA list of integers, along which to reduce. The default is to reduce over all the dimensions of the input tensor. Accepted range is [-r, r-1] where r = rank(data).
Inputs | data : TAn input tensor. | data (differentiable) : TAn input tensor. | axes (optional, non-differentiable) : tensor(int64)Optional input list of integers, along which to reduce. The default is to reduce over all the dimensions of the input tensor if 'noop_with_empty_axes' is false, else act as an Identity op when 'noop_with_empty_axes' is true. Accepted range is [-r, r-1] where r = rank(data).
Outputs | reduced : TReduced output tensor. | reduced (differentiable) : TReduced output tensor. | NO CHANGE
Type constraints | T : tensor(uint32), tensor(uint64), tensor(int32), tensor(int64), tensor(float16), tensor(float), tensor(double), tensor(bfloat16)Constrain input and output types to numeric tensors. | NO CHANGE | NO CHANGE


</body>

</html>

Created table with differences, now making on code changes in relevant files namely src/frontends/onnx/frontend/src/op/reduce.cpp , and src/frontends/onnx/tests/onnx_import.in.cpp .I'm also creating a model for test.

However I'm going for a vacation for a week starting tomorrow, so there will be no updates for a week.

**Comment by gkrivor** [comment]
Hi @mwilczy , thanks for your contribution! I'm only disagree about types, bfloat introduced in opset-13, opset-1 and opset-11 has same list of supported types.

**Comment by inbasperu** [comment]
Hi @mwilczy, just wanted to check if you're still working on this issue. If not, @gkrivor, I'd be keen to take it on if it's available! 
Thanks

**Comment by mlukasze** [comment]
@inbasperu you just took another ticket, please follow that assignment first, if I may ask ;)

**Comment by mlukasze** [comment]
please continue, we can wait a little bit longer ;)

**Comment by p-wysocki** [comment]
Hello @mwilczy, are you still working on that issue? Do you need any help?

**Comment by mwilczy** [comment]
Sure help would be fine, I struggle to find time to complete this, so if
someone want to pick it no problem

**Comment by gkrivor** [comment]
.take
